Firestore/core/test/firebase/firestore/remote/serializer_test.cc

```diff
@@ -221,7 +226,25 @@ TEST_F(SerializerTest, WritesNestedObjectsToBytes) {
                        {{"e", FieldValue::IntegerValue(
                                   std::numeric_limits<int64_t>::max())}})}})}});
 
-  ExpectRoundTrip(model, FieldValue::Type::Object);
+  google::firestore::v1beta1::Value inner_proto;
+  google::protobuf::Map<std::string, google::firestore::v1beta1::Value>* inner_fields = inner_proto.mutable_map_value()->mutable_fields();
+  (*inner_fields)["e"] = EncodeIntegerToProto(std::numeric_limits<int64_t>::max());
+
+  google::firestore::v1beta1::Value middle_proto;
+  google::protobuf::Map<std::string, google::firestore::v1beta1::Value>* middle_fields = middle_proto.mutable_map_value()->mutable_fields();
+  (*middle_fields)["d"] = EncodeIntegerToProto(100);
+  (*middle_fields)["nested"] = inner_proto;
+
+  google::firestore::v1beta1::Value proto;
+  google::protobuf::Map<std::string, google::firestore::v1beta1::Value>* fields = proto.mutable_map_value()->mutable_fields();
+  (*fields)["b"] = EncodeBooleanToProto(true);
```

Comment:
Reading this here I think these can be shorter: maybe even just `ValueProto` with overloads?

I suggest avoiding "Encode" to disambiguate what our serializer is doing vs constructing the expected values.

Reply:
Spoke too soon. linux likes 42L, but mac wants 42LL (due to the way they define int64_t.) Updated to use int64_t{42} instead.